runtime/src/main/kotlin/kotlin/text/Strings.kt

```diff
@@ -7,6 +7,185 @@ package kotlin.text
+ * with the specified [newValue] string.
+ */
+public actual fun String.replaceFirst(oldValue: String, newValue: String, ignoreCase: Boolean): String {
+    val index = indexOf(oldValue, ignoreCase = ignoreCase)
+    return if (index < 0) this else this.replaceRange(index, index + oldValue.length, newValue)
+}
+
+/**
+ * Returns a copy of this string having its first letter lowercased, or the original string,
+ * if it's empty or already starts with a lower case letter.
+ *
+ * @sample samples.text.Strings.decapitalize
+ */
+public actual fun String.decapitalize(): String {
+    return if (isNotEmpty() && this[0].isUpperCase()) substring(0, 1).toLowerCase() + substring(1) else this
+}
+
+/**
+ * Returns `true` if this string is empty or consists solely of whitespace characters.
+ */
+public actual fun CharSequence.isBlank(): Boolean = length == 0 || indices.all { this[it].isWhitespace() }
+
+/**
+ * Returns the substring of this string starting at the [startIndex] and ending right before the [endIndex].
+ *
+ * @param startIndex the start index (inclusive).
+ * @param endIndex the end index (exclusive).
+ */
+@kotlin.internal.InlineOnly
+public actual inline fun String.substring(startIndex: Int, endIndex: Int): String =
+        subSequence(startIndex, endIndex) as String
+
+/**
+ * Returns a substring of this string that starts at the specified [startIndex] and continues to the end of the string.
+ */
+@kotlin.internal.InlineOnly
+public actual inline fun String.substring(startIndex: Int): String =
+        subSequence(startIndex, this.length) as String
+
+/**
+ * Returns `true` if this string starts with the specified prefix.
+ */
+public fun String.startsWith(prefix: String, ignoreCase: Boolean = false): Boolean =
+        regionMatches(0, prefix, 0, prefix.length, ignoreCase)
+
+/**
+ * Returns `true` if a substring of this string starting at the specified offset [startIndex] starts with the specified prefix.
+ */
+public fun String.startsWith(prefix: String, startIndex: Int, ignoreCase: Boolean = false): Boolean =
+        regionMatches(startIndex, prefix, 0, prefix.length, ignoreCase)
+
+/**
+ * Returns `true` if this string ends with the specified suffix.
+ */
+public fun String.endsWith(suffix: String, ignoreCase: Boolean = false): Boolean =
+        regionMatches(length - suffix.length, suffix, 0, suffix.length, ignoreCase)
+
+/**
+ * Returns `true` if the specified range in this char sequence is equal to the specified range in another char sequence.
+ * @param thisOffset the start offset in this char sequence of the substring to compare.
+ * @param other the string against a substring of which the comparison is performed.
+ * @param otherOffset the start offset in the other char sequence of the substring to compare.
+ * @param length the length of the substring to compare.
+ */
+@SymbolName("Kotlin_CharSequence_regionMatches")
+external public actual fun CharSequence.regionMatches(
```

Comment:
Both `Kotlin_CharSequence_regionMatches` and `Kotlin_String_regionMatches` take `KString` parameters. How does this magic works for CharSequences? Or does `KString` correspond to `CharSequence` interface?

Reply:
This is likely a bug, this code doesn't seem to work on non kotlin.String classes.